### PR TITLE
[DUOS-2835][DUOS-2878][risk=no] Use the url dataset property instead of the dbGap version

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2835
https://broadworkbench.atlassian.net/browse/DUOS-2878

## Summary
Use the url dataset property of instead of the dbGap version. The url is meant to replace what we previously stored in dbGap.

See also this UI PR: https://github.com/DataBiosphere/duos-ui/pull/2521
This PR is not blocked on ☝🏽 - both can be moved forward independently.
Production dataset->study conversions necessary for this PR have been completed.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      -- Remove duplicate properties
+      DELETE
+      FROM dataset_property
+      WHERE property_key IN (SELECT key_id FROM dictionary d WHERE KEY = 'dbGap')
+    </sql>
+    <sql>
+      -- Remove deprecated property key from dictionary
+      DELETE
+      FROM dictionary
+      WHERE key = 'dbGap';
```
Comment (fboulnois) [inline src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml]: Do we have datasets that ONLY have dbgap and not url? If so, we should also ensure these are converted over.

**Review by fboulnois — APPROVED** [review]
Overall looks good, one question:

File: src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml (hunk 1 above)
Comment (rushtong) [reply src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml]: On dev, we do. On Prod, we have 3 that need to be migrated. I'll add that detail to the ticket. We cannot move over the old dbgap values - they are not necessarily urls - they were generic strings that didn't go through any validation.

Comment (rushtong) [reply src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml]: I'll also add that we are not using this value anywhere in the UI any longer either - it's essentially dead data.

File: src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+  <changeSet id="changelog-consent-2024-03-28-dataset-dac-id-props.xml" author="grushton">
+    <sql>
+      -- Remove duplicate properties
+      DELETE
+      FROM dataset_property
+      WHERE property_key IN (SELECT key_id FROM dictionary d WHERE KEY = 'dbGap')
```
Comment (aarohinadkarni) [inline src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml]: Looking at what's actually in the database, does this actually need to be the following?
```suggestion
      WHERE property_key IN (SELECT key_id FROM dictionary d WHERE KEY = 'dbGAP')
```

File: src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml (hunk 1 above)
Comment (aarohinadkarni) [inline src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml]: Same here:
```suggestion
      WHERE key = 'dbGAP';
```

File: src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml (hunk 2 above)
Comment (rushtong) [reply src/main/resources/changesets/changelog-consent-2024-04-02-rm-old-dbgap-url.xml]: Nice catch, thanks for that!

**Review by aarohinadkarni — APPROVED** [review]
thanks for the change! 👍

**Pull request merged**